pages/reset-password.vue

```diff
@@ -0,0 +1,188 @@
+<script setup lang="ts">
+import { type Ref, ref } from 'vue'
+import {
+  isPasswordValid,
+  passwordsAreTheSame,
+  renderErrors,
+} from 'lib/form_helpers'
+import { toggleState } from 'lib/helpers'
+import PasswordInput from 'elements/inputs/PasswordInput/PasswordInput.vue'
+import BaseButton from 'elements/buttons/BaseButton/BaseButton.vue'
+import { useChangePassword } from 'composables/api'
+import { useModalAlert } from 'composables/useModalAlert'
+
+definePageMeta({
+  middleware: 'reset-password',
+})
+
+interface RecoverAccountFormState {
+  password: Ref<string>
+  passwordConfirm: Ref<string>
+}
+
+const route = useRoute()
+const resetPasswordCode = route.query?.code as string
+
+const formState: RecoverAccountFormState = {
+  password: ref(''),
+  passwordConfirm: ref(''),
+}
+
+const showPassword = ref(false)
+const errorText = ref('')
+const showErrorsText = ref(false)
+const passwordInputValid = ref(true)
+const passwordConfirmValid = ref(true)
+
+function validatePassword() {
+  passwordInputValid.value = isPasswordValid(formState.password)
+}
+
+function validatePasswordConfirm() {
+  passwordConfirmValid.value =
+    passwordsAreTheSame(formState.password, formState.passwordConfirm) &&
+    isPasswordValid(formState.password)
+}
+
+const { showAlert } = useModalAlert()
+
+async function changePassword() {
+  showErrorsText.value = false
+  await useChangePassword(
+    resetPasswordCode,
+    { password: formState.password.value },
+    {
+      onError: (response: any) => {
+        if (response.error && response.error?.errors) {
```

Comment:
I think it'd be good if we switched this to a `switch(true)` statement makes it quite a bit easier to read

Reply:
I actually wanted to write this differently, but `prettier` yelled at me about it. I'll try it with a `switch` and see how it goes